projects/graph/src/graph.py

```diff
@@ -6,4 +6,85 @@
 class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
     def __init__(self):
-        pass  # TODO
+        self.vertices = {
+            "1": {"2", "3"},
+            "2": {"4", "5", "1"},
+            "3": {"6", "7", "1"},
+            "4": {"2", "8", "9"},
+            "5": {"2", "10", "11"},
+            "6": {"3","12","13"},
+            "7": {"3", "14", "15"},
+            "8": {"4"},
+            "9": {"4"},
+            "10": {"5"},
+            "11": {"5"},
+            "12": {"6"},
+            "13": {"6"},
+            "14": {"7"},
+            "15": {"8"}
+            }
+
+    def add_vertex(self, key):
+        self.vertices[key] = set()
+
+    def add_edge(self, key, value):
+        if not self.vertices[key] and not self.vertices[value]:
+            print('error: no vertext')
+        else:
+            self.vertices[key].add(value)
+            self.vertices[value].add(key)
+
+    def add_directed_edge(self, key, value):
+        if not self.vertices[key] and not self.vertices[value]:
+            print('error: no vertext')
+        else:
+            self.vertices[key].add(value)
+
+    def bf_traverse(self, start):
+        queue = [start]
+        visited = set()
+        while len(queue) > 0:
+            for v in self.vertices[queue[0]]:
+                if v not in queue and v not in visited:
+                    queue.append(v)
+            print(f'VISITED: {queue[0]}')
+            visited.add(queue.pop(0))
+
+    def df_traverse(self, start):
+        stack = [start]
+        discovered = set()
+        # While stack is not empty
+        while len(stack) > 0:
+            v = stack.pop()
```

Comment:
very nice. see how DFT (using a stack) means adding and removing from the same place? you `stack.pop()` any vertex, and then `stack.append()` its neighbors. great job.